### PR TITLE
Filter out `.` files

### PR DESCRIPTION
**Pull request by devpow112**
This allows placing something like a .eslintrc.json file in the folder with lang files.

### CODE REVIEW

**Comment by devpow112** [comment]
@bearfriend hey this is Devon we chatted a few days ago about lang checker stuff. Just integrating with this tool and noticed it would include `.` files when not specifying a set of allowed locales. We had a `.eslintrc.json` file in our `lang/` directories to disable some stuff around quoting. This will just ignore those type of files.

File: bin/cli.js [hunk 1]
```diff
@@ -76,7 +76,9 @@ localesPaths.forEach(localesPath => {
     const sourceLocale = program.sourceLocale || source || globalSource;
     const allowedLocalesString = program.locales || configLocales || globalLocales;
     const allowedLocales = allowedLocalesString && allowedLocalesString.split(',');
-    const filteredFiles = !allowedLocales ? files : files.filter(file => allowedLocales.includes(file.replace(`.${ext}`, '')) || file === sourceLocale + `.${ext}`);
+    const filteredFiles = !allowedLocales ?
+      files.filter(file => !(/(^|\/)\.[^\/\.]/g).test(file)) :
```
Comment (devpow112) [inline bin/cli.js:80]: This should filter out any file starting with a `.` at any path level.

File: bin/cli.js [hunk 2]
```diff
@@ -76,7 +76,9 @@ localesPaths.forEach(localesPath => {
     const sourceLocale = program.sourceLocale || source || globalSource;
     const allowedLocalesString = program.locales || configLocales || globalLocales;
     const allowedLocales = allowedLocalesString && allowedLocalesString.split(',');
-    const filteredFiles = !allowedLocales ? files : files.filter(file => allowedLocales.includes(file.replace(`.${ext}`, '')) || file === sourceLocale + `.${ext}`);
+    const filteredFiles = !allowedLocales ?
+      files.filter(file => !(/^\..*/g).test(file)) :
```
Comment (devpow112) [inline bin/cli.js:80]: Don't include anything that starts with a `.`.

File: bin/cli.js [hunk 3]
```diff
@@ -243,7 +245,7 @@ localesPaths.forEach(localesPath => {
                 break;
               }
 
-              fs.writeFileSync(localePath, newLocaleContents + '\n');
+              fs.writeFileSync(localePath, newLocaleContents + ';\n');
```
Comment (devpow112) [inline bin/cli.js:248]: This is to fix linting issues when using stuff like `eslint`.

File: bin/cli.js [hunk 4]
```diff
@@ -224,15 +226,15 @@ localesPaths.forEach(localesPath => {
               //const contextComment = '';
               switch(format) {
                 case 'js':
-                newLocaleContents = `${exportPrefix} ${formatJS(newLocaleContents)}`;
+                newLocaleContents = `${exportPrefix} ${formatJS(newLocaleContents)};`;
```
Comment (devpow112) [inline bin/cli.js:229]: This, and the subsequent `;`, are just so the generate `js` stuff passes our Brightspace linting. Not sure if you have a preference for tailing `;` or not. I could also mitigate this with `eslint` settings specific to our `lang` folder as long as the code above to ignore `.` style "hidden" files goes in.

Comment (bearfriend) [reply bin/cli.js:229]: No, this makes sense. I was going to say before that it would break or at least not be valid JSON with the previous change.

Comment (devpow112) [reply bin/cli.js:229]: Yeah I realized my previous change was wrong 😞. Glad it makes sense to add it.

File: bin/cli.js (hunk 2 above)
Comment (bearfriend) [reply bin/cli.js:80]: Wow, I didn't realize you could use and test against a regex this way. Neat!

**Review by bearfriend — APPROVED** [review]
There's no versioning or CI set up yet, but I'll try to get this out Monday morning. Thanks!

**Comment by devpow112** [comment]
@bearfriend just want to make sure this wasn't forgotten 🙂

**Comment by bearfriend** [comment]
It absolutely was 🤦

**Comment by devpow112** [comment]
> It absolutely was

No problem 👍

**Pull request merged**